File: frontend/workspace/web/src/api/backend/permission/permissionRule.ts
import createAxios from '/@/utils/axios'

/**
 * 权限规则管理 API 接口
 * 对接 aixone-app-directory 服务的权限规则接口
 * 
 * 注意：所有接口的 tenantId 都从 token 自动获取，不需要显式传递
 */

// 权限规则列表查询参数
export interface PermissionRuleListParams {
    pageNum?: number
    pageSize?: number
    page?: number  // 兼容 baTable 的 page 参数
    limit?: number // 兼容 baTable 的 limit 参数
    pattern?: string
    method?: string
    order?: string
}

// 创建权限规则请求
export interface CreatePermissionRuleRequest {
    pattern: string  // 路径模式，支持Ant路径匹配（**、*）
    methods: string[]  // HTTP方法数组（GET、POST、PUT、DELETE等）
    permission: string  // 权限标识，格式：{resource}:{action} 或 admin:{resource}:{action}
    description?: string
    enabled?: boolean
    priority?: number  // 优先级，数字越大优先级越高
}

// 更新权限规则请求
export interface UpdatePermissionRuleRequest {
    pattern?: string
    methods?: string[]
    permission?: string
    description?: string
    enabled?: boolean
    priority?: number
}

// 权限规则视图
export interface PermissionRuleView {
    id: string
    tenantId: string
    pattern: string
    methods: string[]
    permission: string
    description?: string
    enabled: boolean
    priority: number
    createdAt: string
    updatedAt: string
}

/**
 * 获取权限规则列表（分页，支持过滤）
 * 租户ID从token自动获取
 */
export function getPermissionRuleList(params?: PermissionRuleListParams) {
    return createAxios({
        url: '/api/v1/permissions',
        method: 'get',
        params,
    })
}

/**
 * 获取权限规则详情
 * 租户ID从token自动获取
 */
export function getPermissionRuleById(permissionRuleId: string) {
    return createAxios({
        url: `/api/v1/permissions/${permissionRuleId}`,
        method: 'get',
    })
}

/**
 * 创建权限规则
 * tenantId 从 token 自动获取并设置
 */
export function createPermissionRule(data: CreatePermissionRuleRequest) {
    return createAxios({
        url: '/api/v1/permissions',
        method: 'post',
        data,
    })
}

/**
 * 更新权限规则
 * 租户ID从token自动获取
 */
export function updatePermissionRule(permissionRuleId: string, data: UpdatePermissionRuleRequest) {
    return createAxios({
        url: `/api/v1/permissions/${permissionRuleId}`,
        method: 'put',
        data,
    })
}

/**
 * 删除权限规则
 * 租户ID从token自动获取
 */
export function deletePermissionRule(permissionRuleId: string) {
    return createAxios({
        url: `/api/v1/permissions/${permissionRuleId}`,
        method: 'delete',
    })
}

File: services/aixone-app-directory/src/main/java/com/aixone/directory/permission/domain/aggregate/PermissionRule.java
package com.aixone.directory.permission.domain.aggregate;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 权限规则聚合根
 * 权限规则用于管理接口的权限验证规则（路径-权限映射）
 * 
 * @author AixOne Team
 * @since 1.0.0
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PermissionRule {
    
    private String id;
    private String tenantId;
    private String pattern;  // 路径模式，支持Ant路径匹配（**、*）
    private List<String> methods;  // HTTP方法数组（GET、POST、PUT、DELETE等）
    private String permission;  // 权限标识，格式：{resource}:{action} 或 admin:{resource}:{action}
    private String description;
    private Boolean enabled;
    private Integer priority;  // 优先级，数字越大优先级越高
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    /**
     * 创建权限规则
     */
    public static PermissionRule create(String tenantId, String pattern, List<String> methods, String permission) {
        return PermissionRule.builder()
                .tenantId(tenantId)
                .pattern(pattern)
                .methods(methods)
                .permission(permission)
                .enabled(true)
                .priority(0)
                .createdAt(LocalDateTime.now())
                .updatedAt(LocalDateTime.now())
                .build();
    }
    
    /**
     * 更新权限规则
     */
    public void update(String pattern, List<String> methods, String permission, String description, Boolean enabled, Integer priority) {
        this.pattern = pattern;
        this.methods = methods;
        this.permission = permission;
        this.description = description;
        this.enabled = enabled != null ? enabled : this.enabled;
        this.priority = priority != null ? priority : this.priority;
        this.updatedAt = LocalDateTime.now();
    }
    
    /**
     * 启用权限规则
     */
    public void enable() {
        this.enabled = true;
        this.updatedAt = LocalDateTime.now();
    }
    
    /**
     * 禁用权限规则
     */
    public void disable() {
        this.enabled = false;
        this.updatedAt = LocalDateTime.now();
    }
    
    /**
     * 检查是否属于指定租户
     */
    public boolean belongsToTenant(String tenantId) {
        return this.tenantId != null && this.tenantId.equals(tenantId);
    }
    
    /**
     * 检查是否匹配指定的路径和方法
     */
    public boolean matches(String path, String method) {
        if (!enabled) {
            return false;
        }
        
        // 检查HTTP方法是否匹配
        if (methods != null && !methods.isEmpty() && !methods.contains(method)) {
            return false;
        }
        
        // 检查路径是否匹配（使用Ant路径匹配）
        return matchesPattern(path, pattern);
    }
    
    /**
     * Ant路径匹配
     * 支持 * 和 ** 通配符
     */
    private boolean matchesPattern(String path, String pattern) {
        if (path == null || pattern == null) {
            return false;
        }
        
        // 完全匹配
        if (path.equals(pattern)) {
            return true;
        }
        
        // 转换为正则表达式
        String regex = pattern
                .replace(".", "\\.")
                .replace("**", ".*")
                .replace("*", "[^/]*");
        
        return path.matches(regex);
    }
}

